tests/integration/test_prometheus_integration.py

```diff
@@ -39,14 +41,31 @@ async def labels(self) -> List[str]:
         return result["data"] if result["status"] == "success" else []
 
 
+@pytest.fixture
+def updated_charmlib():
```

Comment:
consider reusing [traefik's spellbook](https://github.com/canonical/traefik-k8s-operator/tree/main/tests/integration/spellbook) to get rid of this boilerplate

Reply:
Nice! I'd postpone it for a later PR, want to minimize the changes in this PR so I can land it ASAP. Thanks!